dart/lib/src/default_integrations.dart

```diff
@@ -13,22 +13,59 @@ class RunZonedGuardedIntegration extends Integration {
 
   final Future<void> Function() _runner;
 
+  /// Needed to check if we somehow caused a `print()` recursion
+  bool _isPrinting = false;
```

Comment:
this might need to be an atomic var if print could be called concurrently across different threads (isolates)?

Reply:
not even sure if it'd be called again from a different isolate anyway, nor I know if Dart has Atomic vars, food for thought though

Reply:
Because this is private you can't set this in any other Isolate. Different Isolates have different Zones, so this won't be called.